diff --git a/samples/javascript_nodejs/02.a.echobot/bot.js b/samples/javascript_nodejs/02.a.echobot/bot.js
index 845e5156..d6881f5e 100644
--- a/samples/javascript_nodejs/02.a.echobot/bot.js
+++ b/samples/javascript_nodejs/02.a.echobot/bot.js
@@ -11,9 +11,11 @@ class MyBot {
     async onTurn(turnContext) {
         // See https://aka.ms/about-bot-activity-message to learn more about the message and other activity types.
         if (turnContext.activity.type === ActivityTypes.Message) {
-            await turnContext.sendActivity(`You said '${ turnContext.activity.text }'`);
+            let txt = `You said '${ turnContext.activity.text }'`;
+            await turnContext.sendActivity( { text: txt, speak: txt, inputHint: 'acceptingInput' } );
         } else {
-            await turnContext.sendActivity(`[${ turnContext.activity.type } event detected]`);
+            let txt = `[${ turnContext.activity.type } event detected]`;
+            await turnContext.sendActivity( { text: txt, speak: txt, inputHint: 'acceptingInput' } );
         }
     }
 }
